File: 단기43차/Java(eclipse)/0529/src/Student.java
/*
 * page 344, 257(Wrapper 클래스 : 멤버변수로  value형 타입을 갖는 클래스)
 * 
 *  만약 학벙을 저장하고자 한다면??
 * - 학번의 타입이 정해져 있지 않다면??(String or int)
 * -1) 대안 : Object
 */
public class Student {
	private Object number;
	
	public void set(Object value) {
		number = value;
	}
	
	public Object get() {
		return number;
		
	}
}

File: 단기43차/C언어/참고(짝꿍)/C 언어/0416/0416/student.cpp
//student.cpp

#include <iostream>
using namespace std;

#include "student.h"

student::student(int _number, string _name, string _major)
{
	number = _number;
	name = _name;
	major = _major;
}


student::~student()
{
}


int student::getnumber()
{
	return number;
}


string student::getname()
{
	return name;
}


string student::getmajor()
{
	return major;
}


void student::setmajor(string value)
{
	major = value;
}


void student::printline()
{
	cout << number << "\t";
	cout << name.c_str() << "\t";
	cout << major.c_str() << endl;
}


void student::print()
{
	cout << "학번 : " << number << endl;
	cout << "이름 : " << name.c_str() << endl;
	cout << "학과 : " << major.c_str() << endl;
}
